commands/platform.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+var fs = require('fs'),
+    path = require('path');
+
+var Q = require('q');
+
+var lib = require('manifoldjs-lib');
+
+var log = lib.log,
+    platformTools = lib.platformTools,
+    utils = lib.utils;
+
+// registers a new platform module    
+function addPlatform (program) {
+  if (program.args.length < 3) {
+    return Q.reject(new Error('You must specify a platform ID.'));    
+  }
+  
+  if (program.args.length < 4) {
+    return Q.reject(new Error('You must specify a package source for the platform. This can be an npm package, a GitHub URL, or a local path.'));    
+  }
+  
+  var platformId = program.args[2].toLowerCase();
+  var source = program.args[3];
+  
+  return platformTools.addPlatform(platformId, source).then(function () {
+    log.info('The \'' + platformId + '\' platform was registered successfully.');      
+  });
+}
+
+// removes a registered platform module
+function removePlatform (program) {

```

Comment:
All commands receive a program object with command line arguments. This allows extending a command by adding a new parameter without having to make too many changes. For example, removePlatform currently de-registers the specified platform, but one of the things that we might implement in the future is a switch to delete corresponding platform module from disk (i.e. _manifoldjs platform remove <plaformId> [-delete]_).